parser/parser/cc/grammars/typedruby.ypp

```diff
@@ -909,26 +924,31 @@ int yylex(parser::semantic_type *lval, ruby_parser::location *lloc, ruby_parser:
                     }
 
 lbrace_cmd_block_start:
-                  tLBRACE_ARG { driver.lex.context.push(Context::State::BLOCK); $$ = $1; }
+                  tLBRACE_ARG
+                    {
+                      driver.lex.context.inBlock = true;
+                      driver.lex.context.allowNumparams = true;
+                      $$ = driver.alloc.token_with_context($1, driver.lex.context.dup());
+                    }
  cmd_brace_block: lbrace_cmd_block_start brace_body tRCURLY
                     {
                       auto &block = $2;
-                      block->begin = $1;
+                      block->begin = $1->token;
                       block->end = $3;
                       $$ = block;
-                      driver.lex.context.pop();
+                      driver.lex.context.inBlock = $1->context.inBlock;
                     }
                   // Error-recovery case for typos like "foo {&:bar}"
                 | lbrace_cmd_block_start tAMPER arg_value tRCURLY
                     {
                       auto args = driver.build.args(self, nullptr, driver.alloc.node_list(), nullptr, false);
                       auto block = driver.alloc.delimited_block(nullptr, args, $3, nullptr);
-                      block->begin = $1;
+                      block->begin = $1->token;
                       block->end = $4;
                       $$ = block;
                       driver.diagnostics.emplace_back(dlevel::ERROR, dclass::CurlyBracesAroundBlockPass,
                           diagnostic::range(@1.beginPos(), @4.endPos()));
-                      driver.lex.context.pop();
+                      /* driver.lex.context.inBlock = $2.context.inBlock; */
```

Comment:
ditto, seems like maybe this is dead code?

Reply:
unclear, because the previous production rule has very similar looking code but it's not commented out?